File: frontend/types/festival.ts
export interface EnergySource {
  name: string;
  capacity_kWh: number;
  share_percent: number;
}

export interface EnergyBreakdown {
  total_supply_kWh: number;
  total_demand_kWh: number;
  sources: EnergySource[];
  solar_panels_count: number;
  solar_area_m2: number;
  battery_units: number;
  battery_capacity_kWh: number;
  hourly_solar_kWh: Record<string, number>;
  hourly_prices_eur_mwh: Record<string, number>;
}

export interface ResourceBreakdown {
  water_liters_total: number;
  water_liters_per_person: number;
  food_kg_total: number;
  food_kg_per_person: number;
  drinks_kg_total: number;
  drinks_kg_per_person: number;
  waste_kg_total: number;
  waste_kg_per_person: number;
  toilets_required: number;
  human_waste_kg: number;
}

export interface CostEstimate {
  solar_panels_eur: number;
  batteries_eur: number;
  total_eur: number;
}

export interface FestivalPlan {
  max_visitors: number;
  location_name: string;
  area_m2: number;
  duration_days: number;
  month: number;
  energy: EnergyBreakdown;
  resources: ResourceBreakdown;
  cost: CostEstimate;
  grid_congested: boolean;
  renewable_percent: number;
}

export interface PlanRequest {
  lat: number;
  lng: number;
  area_m2: number;
  duration_days: number;
  month: number;
}

File: backend/models/festival.py
from pydantic import BaseModel, Field


class EnergySource(BaseModel):
    name: str
    capacity_kWh: float
    share_percent: float


class EnergyBreakdown(BaseModel):
    total_supply_kWh: float
    total_demand_kWh: float
    sources: list[EnergySource]
    solar_panels_count: int
    solar_area_m2: float
    battery_units: int
    battery_capacity_kWh: float
    hourly_solar_kWh: dict[str, float] = Field(default_factory=dict, description="Hour → kWh solar output")
    hourly_prices_eur_mwh: dict[str, float] = Field(default_factory=dict, description="Hour → EUR/MWh spot price")


class ResourceBreakdown(BaseModel):
    water_liters_total: float
    water_liters_per_person: float
    food_kg_total: float
    food_kg_per_person: float
    drinks_kg_total: float
    drinks_kg_per_person: float
    waste_kg_total: float
    waste_kg_per_person: float
    toilets_required: int
    human_waste_kg: float


class CostEstimate(BaseModel):
    solar_panels_eur: float
    batteries_eur: float
    total_eur: float


class FestivalPlan(BaseModel):
    max_visitors: int
    location_name: str
    area_m2: float
    duration_days: int
    month: int
    energy: EnergyBreakdown
    resources: ResourceBreakdown
    cost: CostEstimate
    grid_congested: bool = Field(description="Whether the local grid is at capacity")
    renewable_percent: float = Field(description="Percentage of energy from renewables")
